### DISCUSSION TITLE
Is it possible to configure NX to run a mock server as a pre-requisit to an application?

### DISCUSSION THREAD

**Discussion by nemonemi** [nrwl/nx#7803] [General]
I have a mock server that I want to run as a precondition to the application.
Also, when I run e2e tests for that application, the same server should be running together with the application.

How to achieve this?

I have tried using the `dependsOn` and a custom script, but the problem is that it would only run that custom script content since it's a long running process (a running server), and aside from that, it doesn't deal with the teardown of the mock server.

**Comment by dimitriy-k** [comment]
is there any solution nowadays?

**Comment by nemonemi** [comment]
Hello @dimitriy-k ,
I found the solution, and have been happily using it for the past ~2 years.

I use the [MSW](https://mswjs.io/) for my mocking solution, and I have it configured to run when the application starts in development mode.
When the Cypress tests start, they run the application in development mode. As the frontend server starts, so does the mock server, and thus the tests assert the application features against the mock server.

**Reply by ajitzero** [reply]
Hi @nemonemi, is there any Nx generator for this?

**Reply by nemonemi** [reply]
Not at the moment. However, the implementation is not complicated.
The way I've set it up is, I have a library containing MSW resolvers, data, schema, etc. and I import it in the main file of the application.  There, you can check on what stage your app is on so that it runs only locally if that is what you wish. 

That means that when you run tests locally, and when you start your e2e tests server, the state will be the same.
